pkg/provision/engine.go

```diff
@@ -1162,3 +1189,57 @@ func (e *NativeEngine) GetWorkloadStatus(id string) (gridtypes.ResultState, bool
 
 	return wl.Result.State, true, nil
 }
+
+// getTwinVerificationStatus make sure the account used is verified we have the user public key in bytes(pkBytes)
+func getTwinVerificationStatus(twinID uint32) (status string) {
+	status = "FAILED"
+	env := environment.MustGet()
+
+	verificationServiceURL, err := url.JoinPath(env.KycURL, "/api/v1/status")
+	if err != nil {
+		return
+	}
+
+	request, err := http.NewRequest(http.MethodGet, verificationServiceURL, nil)
+	if err != nil {
+		return
+	}
+
+	q := request.URL.Query()
+	q.Set("twin_id", fmt.Sprint(twinID))
+	request.URL.RawQuery = q.Encode()
+
+	cl := &http.Client{
+		Timeout: 10 * time.Second,
+	}
+
+	response, err := cl.Do(request)
+	if err != nil {
+		return
+	}
+	defer response.Body.Close()
+
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		return
+	}
+
+	var result struct {
+		Result struct {
+			Status string `json:"status"`
+		} `json:"result"`
+		Error string `json:"error"`
+	}
+
+	err = json.Unmarshal(body, &result)
+	if err != nil {
+		return
+	}
+
+	if response.StatusCode != http.StatusOK {
+		log.Error().Msgf("failed to verify user status: %s", result.Error)
+		return
+	}
```

Comment:
This check needs to be done before you actually deserialize the body, no? If status is not okay a big chance the body is not json